### PR TITLE
New docs folder in starcellbio_html develop branch

### PR DESCRIPTION
**Pull request by cshubert**
This includes lots of StarCellBio documentation as well as a new Developer's Guide.

### CODE REVIEW

File: docs/DeveloperGuide.rst [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    :download:`StarCellBio Architecture <StarCellBio.pdf>`
+
+Create a Virtual Environment for developing StarCellBio
+-------------------------------------------------------
+"A Virtual Environment ... is an isolated working copy of Python which allows
+you to work on a specific project without worry of affecting other project."
+
+Reference: `http://docs.python-guide.org/en/latest/dev/vertualenvs/ <http://docs.python-guide.org/en/latest/dev/vertualenvs/>`_
+
+This sample expects the ``PyVENV`` folder to contain the Virtual Environment for the sample.
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:31]: maybe change this to something more specific, I have a lot of venvs and something more descriptive like `starcellbio_venv` may be more helpful

File: docs/DeveloperGuide.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    (PyVENV) $ sudo launchctl load -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To Stop mysql::
+
+    (PyVENV) $ sudo launchctl unload -w ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
+
+To pip install
+
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:102]: you don't want to do this as sudo since that won't install into the venv

File: docs/DeveloperGuide.rst [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    Install the database drivers if prompted
+
+
+mysqld command starts database
+
+Reference: `Installing MySQL on Mac OS X <https://rtcamp.com/tutorials/mac/osx-brew-php-mysql-nginx/>`_
+
+If you need to install MySQL on Mac OS X, execute the following command::
+
+    $ brew install mysql --enable-debugging
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:82]: I think we should consider adding a Vagrantfile to the repo for development.  That will make setting up the development environment easier and we can control the packages/not deal with instructions for a lot of different systems

File: docs/DeveloperGuide.rst [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+In new terminal window execute the following commands::
+
+    $ source ~/PyVENV/bin/activate
+    (PyVENV) $ cd starcellbio_html
+    (PyVENV) $ sudo pip install -r requirements.txt
+
+    (PyVENV) $ export PROJECT_HOME="/Users/starcellbio/starcellbio_html" # watch.py needs repo at PROJECT_HOME
+    (PyVENV) $ cd $PROJECT_HOME/html_app # watch.py needs repo in this location
+    (PyVENV) $ python watch.py
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:121]: watch.py requires [MacFSEvents](https://pypi.python.org/pypi/MacFSEvents) which is currently commented out on requirements.txt.  It also requires a Mac, and I use linux for development, so this is a problem

Comment (carsongee) [inline docs/DeveloperGuide.rst:121]: We should probably port this to [watchdog](https://pypi.python.org/pypi/watchdog)

File: docs/DeveloperGuide.rst [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+.. note:: Django may not work due to an error starting mysql. The first two lines that follow get mysql up
+and running. The second two lines start the StarCellBio Django server. If you don't have mysql installed,
+you'll need to do that.
+
+
+    (PyVENV) $ cd $PROJECT_HOME
+
+    (PyVENV) $ sudo ./manage.py collectstatic
+    (PyVENV) $ sudo ./manage.py loaddata backend statuses courses assignments studentassignments
+    (PyVENV) $ sudo ./manage.py runserver
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:139]: again, not sudo here

Comment (carsongee) [inline docs/DeveloperGuide.rst:139]: Also, and this is part of where you will run into trouble (after you get MySQL running), the default username and password need to be created in MySQL.  That appears to be `starcellbio` and `136a411ed9e8592089444b7164ffaf84` for the database `starcellbio`.  We can either do this something like an ansible play, or simpler with just the command:

``` shell
echo "CREATE DATABASE starcellbio; GRANT ALL PRIVILEGES ON starcellbio.* TO 'starcellbio'@'localhos\
t' IDENTIFIED BY '136a411ed9e8592089444b7164ffaf84';" | sudo mysql
```

if they don't have a root user password for mysql

File: docs/DeveloperGuide.rst [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+./manage.py loaddata backend statuses courses assignments studentassignments
```
Comment (carsongee) [inline docs/DeveloperGuide.rst:141]: So I think you need a:

```
./manage.py syncdb
```

which will work with 1.4.2, but updating to the most secure version of Django 1.4 I get:

```
Problem installing fixture '/vagrant/backend/fixtures/assignments.xml': Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py", line 190, in handle
    for obj in objects:
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/serializers/xml_serializer.py", line 162, in next
    for event, node in self.event_stream:
  File "/usr/lib/python2.7/xml/dom/pulldom.py", line 232, in next
    rc = self.getEvent()
  File "/usr/lib/python2.7/xml/dom/pulldom.py", line 265, in getEvent
    self.parser.feed(buf)
  File "/usr/lib/python2.7/xml/sax/expatreader.py", line 210, in feed
    self._parser.Parse(data, isFinal)
  File "/home/vagrant/.virtualenvs/starcellbio/local/lib/python2.7/site-packages/django/core/serializers/xml_serializer.py", line 317, in start_doctype_decl
    raise DTDForbidden(name, sysid, pubid)
DTDForbidden: DTDForbidden(name='django-objects', system_id=None, public_id=None)
```

which is something we will want to investigate.

Comment (carsongee) [inline docs/DeveloperGuide.rst:141]: After all that, I have it up and running, and it looks like:
image

which seems right, but I really have no idea.

**Comment by pdpinch** [comment]
I think we got what we needed out of this.

**Pull request closed without merging**